database/challenge.go

```diff
@@ -204,6 +232,18 @@ func (cr *ChallengeResponse) LoadFromRequest(req *http.Request) error {
 	return nil
 }
 
+// LoadFromReader loads a ChallengeResponse from the given io.Reader.
+//
+// Typically, this reader will be a request.Body.
+func (cr *ChallengeResponse) LoadFromReader(r io.Reader) error {
+	// Parse the request's body.
+	b, err := ioutil.ReadAll(r)
```

Comment:
f/u It's not obvious that `ReadAll` is limited by a `LimitReader`. So I'd apply the limit reader in here.